sfa_dash/api_interface/cdf_forecast_groups.py

```diff
@@ -1,14 +1,36 @@
+import pandas as pd
+
+
 from sfa_dash.api_interface import get_request, post_request, delete_request
+from sfa_dash.api_interface.cdf_forecasts import get_values as get_cdf_values
+from sfa_dash.api_interface.cdf_forecasts import valid_times as cdf_valid_times
+from solarforecastarbiter.io.utils import json_payload_to_forecast_series
 
 
 def get_metadata(forecast_id):
     req = get_request(f'/forecasts/cdf/{forecast_id}')
     return req
 
 
-def get_values(forecast_id, **kwargs):
-    req = get_request(f'/forecasts/cdf/{forecast_id}/values', **kwargs)
-    return req
+def get_values(metadata, **kwargs):
+    """
+    Parameters
+    ----------
+    metadata: dict
+        The metadata dictionary of a probabilistic forecast group as returned
+        from the API.
+
+    Returns
+    -------
+    pandas.DataFrame
+        Dataframe where column names are constant values.
+    """
+    constant_values = metadata['constant_values']
+    data = pd.DataFrame()
+    for cv in constant_values:
+        req = get_cdf_values(cv["forecast_id"], **kwargs)
+        data[str(cv['constant_value'])] = json_payload_to_forecast_series(req)
```

Comment:
I wonder about replacing this with ``concurrent.futures.ThreadPoolExecutor.map`` or similar since it will mostly be waiting for the api to respond and there might be a bunch of constant valvues

Reply:
I don't recall what goes into get_request or if threading will be a problem for the session

Reply:
Alright, I got this with a wrapper function that copies the current app context, and using flasks `copy_current_request_context` wrapper. Exceptions should be reraised when trying to access responses from the iterator.

Reply:
Cool! sorry, didn't think about the flask request context for the oauth session etc.

Will something like this work to continue processing while some threads are waiting for responses?
```python
def request_constant_value_values(cv):
    return cv['constant_value'], json_payload_to_forecast_series(get_cdf_values(cv["forecast_id"], **kwargs))

all_series = ...
return pd.DataFrame.from_items(all_series)
```

Reply:
I'm not sure if what I've done in this update will have the intended affect. I think what you're going for is to pass the generator before the futures are complete? I'm not super familiar with concurrent.futures, but I did see that something I missed in my first stab at this was shutting down the executor, so I moved things into a with statement so we don't leak threads.

Reply:
I think it does what I was getting at, making the request and processing the json into a series in each thread as the data comes in, then gather all the series to make the dataframe